generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts

```diff
@@ -1042,7 +1042,10 @@ export class GeneratedSdkClientClassImpl implements GeneratedSdkClientClass {
     public getOptionsPropertiesForSnippet(context: SdkContext): ts.ObjectLiteralElementLike[] {
         const properties: ts.ObjectLiteralElementLike[] = [];
 
-        if (!this.requireDefaultEnvironment && context.ir.environments?.defaultEnvironment == null) {
+        if (
+            !this.requireDefaultEnvironment &&
+            context.environments.getGeneratedEnvironments().hasDefaultEnvironment()
+        ) {
```

Comment:
@petesiki curious what led you to this being the fix?

Reply:
(i would look here https://github.com/fern-api/fern/blob/eff31d23aaa180caee48d55e056c3ca703b41c55/generators/typescript/sdk/client-class-generator/src/GeneratedSdkClientClassImpl.ts#L1282)

Reply:
thanks @dsinghvi